src/never.rs

```diff
@@ -0,0 +1,91 @@
+use {Future, Poll};
+
+/// A future which is never resolved.
+///
+/// This future can be created with the `empty` function.
+#[derive(Copy, Clone)]
+pub struct Never {}
+
+/// Creates a future which never resolves, representing a computation that never
+/// finishes.
+///
+/// The returned future will never resolve with a success but is still
+/// susceptible to cancellation. That is, if a callback is scheduled on the
+/// returned future, it is only run once the future is dropped (canceled).
+pub fn never<T, E>() -> impl Future<Item = T, Error = E> + Send + 'static {
```

Comment:
Nope! I just want to use `!` as a type. `impl Trait` was a way for me to write the convenience `never` function (which can return a `Future` of with any `Item` and `Err` types) with the crazy `MapErr<Map...>` type. If we just make that function return `Never`, then there's no use for `impl Trait`, but then there should probably be convenience methods on `Future<!, E>`, `Future<T, !>`, and `Future<!, !>` to allow the inner `!` to coerce.